Routes cart #73

clocasto opened the pull request:
Large overhaul of all cart and order routes, DB handling, and factories. Should be able to manage carts and orders without error (except duplicate items in a cart -> not sure how to support that).

Also wrote barebones testing script for carts. All server testing except for this file is disabled because modifying any server file would cause the development database to drop.

clocasto committed:
fixed a lot of cart/order routes & DB logic

clocasto committed:
further fixes to cart routing EXCEPT updating cart status

clocasto committed:
fixed all routes/DB handling for carts

clocasto committed:
completed all cart/order routes and factories

nrod80 reviewed (approved):
this looks really solid, nice work. I'm going to approve, but I do have two quick things: Remind me what it means for a cart to be 'active' and what is the .addScope in the associations?

clocasto committed:
refactoring of local authentication

clocasto committed:
Update gulpfile.js

Removing changes to server test tasks in the gulpfile - probably unnecessary to seed the database prior to testing (and the watch was not properly triggering a reseed).

curiousYi commented on browser/js/factories/orders.factory.js:
```diff
@@ -0,0 +1,25 @@
+app.factory('CartFactory', function($http) {
```
I think this should be OrdersFactory?

clocasto replied:
Good catch!

curiousYi reviewed (commented):
I think the factory in browser/js/factories/orders.factory.js is misnamed.

clocasto committed:
Update orders.factory.js

Correcting typo - should be OrderFactory
